draft-ietf-rats-corim.md

```diff
@@ -387,8 +387,18 @@ The following describes each child element of this type.
 
 ### Profile Types {#sec-corim-profile-types}
 
-A profile specifies which of the optional parts of a CoRIM are required, which
-are prohibited and which extension points are exercised and how.
+Profiling is the mechanism that allows the base CoRIM schema to be customised to fit a specific Attester or Attesters' family.
+
+A profile defines which of the optional parts of a CoRIM are required, which are prohibited and which extension points are exercised and how.
+A profile MUST NOT alter the syntax or semantics of an already defined CoRIM type.
+A profile MAY constrain the values of a given CoRIM type to a subset of that type.
```

Comment:
```suggestion
A profile MAY constrain the set of a given CoRIM type to a subset of that type.
A profile MAY extend the set of a given CoRIM type to meet the requirement of a specific profile.
```

Reply:
I think 'values' is more intuitive than 'set' as set implies the type structure follows mathematical sets. But 'subset' isn't inferring mathematical sets.

Reply:
I felt it a bit like key value semantics hence suggested, with the set.
Perhaps `choice` may fit better, isn't it?

Reply:
maybe less is more?

Reply:
"set" alone doesn't sound good.  If you were instead suggesting "set of values", I concur with Ned that "values" alone does the job.

Reply:
ok, fine. However what about  the other aspect of extend also ? It seems to say that we only Constrain the types.

Reply:
I suppose the native types which are `$type-choices` can be extended, isn't it ?

Reply:
Not necessarily, only when we say so.

Reply:
> Not necessarily, only when we say so.

In the context of defining profile rules, the profile mechanism is a way to extend / restrict the standard (base) schema in a non-standard way. By requiring the profile to be public, there is an approximation of a 'defacto' standardization. This flexibility is desirable even necessary as a way to accommodate the wide range of product differences. However, profiles are better than fully proprietary approaches. That is the value that should be highlighted. Profiles may be useful for identifying features that can / should be adopted into a standard / base schema over time. I think this is what is meant by "only when we say so".

Reply:
> ok, fine. However what about the other aspect of extend also ? It seems to say that we only Constrain the types.

CDDL is a language of 'types'. Even maps and records are typing information. A schema is a type structure that is used to check a data structure for type compliance. It's OK to refer to the schema as a collection of 'type' structures IMHO. Otherwise, its ok to use 'schema', as in: "A profile MAY constrain or extend CoRIM schema statements."